core/helpers/router.ts

```diff
@@ -1,39 +1,52 @@
-import VueRouter, { RouteConfig } from 'vue-router'
-import { RouterManager } from '@vue-storefront/core/lib/router-manager';
+import VueRouter, { RouteConfig } from "vue-router";
+import { RouterManager } from "@vue-storefront/core/lib/router-manager";
+import { ErrorHandler, RawLocation, Route } from "vue-router/types/router";
+
+const originalPush = VueRouter.prototype.push;
+VueRouter.prototype.push = function push(
+  location: RawLocation,
+  onComplete: Function = () => {},
+  onAbort?: ErrorHandler
+): Promise<Route> {
+  if (onComplete || onAbort)
+    return originalPush.call(this, location, onComplete, onAbort);
+  return originalPush.call(this, location).catch(err => err);
+};
```

Comment:
last thing :) `'vue-router'` module is shared between requests so it will add this new function to prototype with every request. We don't need to do that. There is helper that will add it only once, it is called `once` :D Can you wrap all of this with `once`?
`import { once } from '@vue-storefront/core/helpers'`
```
once('__VUE_EXTEND_PUSH_RR__', () => {
  ...
})
```

Reply:
Pushed the fix. Sorry for all the mistakes on my site. this is actually my first pull request :D

Reply:
thanks for your effort :) it looks great